bitcoin/src/address/mod.rs

```diff
@@ -157,27 +169,27 @@ impl Payload {
 
     /// Creates a pay to script hash P2SH payload from a script
     #[inline]
-    pub fn p2sh(script: &Script) -> Result<Payload, Error> {
+    pub fn p2sh(script: &Script) -> Result<Payload, PayloadError> {
         if script.len() > MAX_SCRIPT_ELEMENT_SIZE {
-            return Err(Error::ExcessiveScriptSize);
+            return Err(PayloadError::ExcessiveScriptSize);
         }
         Ok(Payload::ScriptHash(script.script_hash()))
     }
 
     /// Create a witness pay to public key payload from a public key
-    pub fn p2wpkh(pk: &PublicKey) -> Result<Payload, Error> {
+    pub fn p2wpkh(pk: &PublicKey) -> Result<Payload, PayloadError> {
```

Comment:
I was expecting this to return only `UncompressedPubKeyError`, not the entire `PayloadError`.

Reply:
Wow, you are totally correct and this might not be a good module to test run this new error convention on because loads of the `Payload` functions need changing. I'm going to convert this to draft and go find an easier module to proof this now convention with.